### ISSUE TITLE
Error getting symbol file: IOException: closed

### ISSUE THREAD

**Issue by justanotheranonymoususer** [NationalSecurityAgency/ghidra#3184]
**Describe the bug**
I'm trying to download a pdb file, and it fails. I tried several times. Downloads fine with the browser.
**Also:** It failed to be downloaded during analysis, and I got no indication. In addition, when I tried to download it manually, I got an error saying that analysis was already completed and that it's not optimal. So perhaps, instead of silently proceeding with analysis, warn the user and give the chance to retry the download.

**To Reproduce**
1. Load [this file](https://msdl.microsoft.com/download/symbols/explorerextensions.dll/60C47247444000/explorerextensions.dll) in Ghidra.
2. Download PDB: 
![image](https://user-images.githubusercontent.com/5781692/124355374-6e6f1980-dc19-11eb-9a9c-cc24bae6a8f5.png)
3. See error: 
![image](https://user-images.githubusercontent.com/5781692/124355381-762ebe00-dc19-11eb-84d0-f76167e20dc1.png)

**Expected behavior**
No error.

**Environment (please complete the following information):**
 - Win10
 - Java Version: jdk11.0.7_10
 - Ghidra Version: 10.0
 - Ghidra Origin: official ghidra-sre.org distro

**Additional context**
```
java.io.IOException: closed
---------------------------------------------------
Build Date: 2021-Jun-21 1239 EDT
Ghidra Version: 10.0
Java Home: C:\Program Files\Amazon Corretto\jdk11.0.7_10
JVM Version: Amazon.com Inc. 11.0.7
OS: Windows 10 10.0 amd64
```

**Comment by dev747368** [comment]
Could you check your symbols directory and see if the ExplorerExtensions.pdb got successfully downloaded into its subdirectory?

I'm not seeing your exact error message, but I am seeing something weird where the download works but the analyzer is failing.  When I try the process again, it shows the pdb in its local directory (ie. not remote anymore), but again the analyzer is having an issue.

Also, could you check your ghidra log file and post any stack traces that may have preceded this?

**Comment by justanotheranonymoususer** [comment]
> Could you check your symbols directory and see if the ExplorerExtensions.pdb got successfully downloaded into its subdirectory?

No, it is being removed:

![8nRlpse0lM_2](https://user-images.githubusercontent.com/5781692/124366404-4e127f80-dc58-11eb-9a6a-ca0e368faefc.gif)

> Also, could you check your ghidra log file and post any stack traces that may have preceded this?

```
2021-07-03 23:49:45 INFO  (DefaultProject) Opening project: C:\proj  
2021-07-03 23:50:06 INFO  (PackedDatabaseCache) Packed database cache: C:\Users\x\AppData\Local\Ghidra\packed-db-cache  
2021-07-03 23:50:06 DEBUG (PackedDatabaseCache) Using cached packed database: C:\ghidra-data\typeinfo\winapi_64.gdt  
2021-07-03 23:50:06 DEBUG (PackedDatabaseCache) Using cached packed database: C:\ghidra-data\typeinfo\ntdiff_ntoskrnl_ntdll_hal_Win10_1903_19H1_x64.gdt  
2021-07-03 23:50:06 DEBUG (PackedDatabaseCache) Using cached packed database: C:\ghidra_10.0_PUBLIC\Ghidra\Features\Base\data\typeinfo\win32\windows_vs12_64.gdt  
2021-07-03 23:50:06 DEBUG (PackedDatabaseCache) Using cached packed database: C:\ghidra-data\typeinfo\ntddk_64.gdt  
2021-07-03 23:50:23 DEBUG (SymbolServerService) SymbolServerService: querying C:\ProgramData\Dbg\sym for ExplorerExtensions.pdb, cd9e6ca1-215b-466a-9ca7-d946299de6e1, 1, 0, ???  
2021-07-03 23:50:23 DEBUG (LocalSymbolStore) Detected symbol file directory: C:\ProgramData\Dbg\sym\advapi32.pdb  
2021-07-03 23:50:23 WARN  (LocalSymbolStore) Symbol directory missing control files, guessing storage scheme as level 1: C:\ProgramData\Dbg\sym  
2021-07-03 23:50:23 DEBUG (SymbolServerService) SymbolServerService: got 0 results from C:\ProgramData\Dbg\sym  
2021-07-03 23:50:23 DEBUG (SymbolServerService) SymbolServerService: skipping non-local symbol server https://msdl.microsoft.com/download/symbols/  
2021-07-03 23:50:23 DEBUG (SymbolServerService) SymbolServerService: found 0 matches  
2021-07-03 23:50:33 DEBUG (SymbolServerService) SymbolServerService: querying C:\ProgramData\Dbg\sym for ExplorerExtensions.pdb, cd9e6ca1-215b-466a-9ca7-d946299de6e1, 1, 0, ???  
2021-07-03 23:50:33 DEBUG (SymbolServerService) SymbolServerService: got 0 results from C:\ProgramData\Dbg\sym  
2021-07-03 23:50:33 DEBUG (SymbolServerService) SymbolServerService: querying https://msdl.microsoft.com/download/symbols/ for ExplorerExtensions.pdb, cd9e6ca1-215b-466a-9ca7-d946299de6e1, 1, 0, ???  
2021-07-03 23:50:33 DEBUG (HttpSymbolServer) HttpSymbolServer[https://msdl.microsoft.com/download/symbols/]: Checking exist for [index2.txt]: https://msdl.microsoft.com/download/symbols/index2.txt HEAD  
2021-07-03 23:50:35 DEBUG (HttpSymbolServer) HttpSymbolServer[https://msdl.microsoft.com/download/symbols/]: Response: 404  
2021-07-03 23:50:35 DEBUG (HttpSymbolServer) HttpSymbolServer[https://msdl.microsoft.com/download/symbols/]: Checking exist for [ExplorerExtensions.pdb/CD9E6CA1215B466A9CA7D946299DE6E11/ExplorerExtensions.pdb]: https://msdl.microsoft.com/download/symbols/ExplorerExtensions.pdb/CD9E6CA1215B466A9CA7D946299DE6E11/ExplorerExtensions.pdb HEAD  
2021-07-03 23:50:37 DEBUG (HttpSymbolServer) HttpSymbolServer[https://msdl.microsoft.com/download/symbols/]: Response: 200  
2021-07-03 23:50:37 DEBUG (SymbolServerService) SymbolServerService: got 1 results from https://msdl.microsoft.com/download/symbols/  
2021-07-03 23:50:37 DEBUG (SymbolServerService) SymbolServerService: found 1 matches  
2021-07-03 23:50:39 DEBUG (SymbolServerService) SymbolServerService: getting symbol file: https://msdl.microsoft.com/download/symbols/ExplorerExtensions.pdb/CD9E6CA1215B466A9CA7D946299DE6E11/ExplorerExtensions.pdb  
2021-07-03 23:50:39 DEBUG (SymbolServerService) SymbolServerService: copying file https://msdl.microsoft.com/download/symbols/ExplorerExtensions.pdb/CD9E6CA1215B466A9CA7D946299DE6E11/ExplorerExtensions.pdb from remote to local C:\ProgramData\Dbg\sym  
2021-07-03 23:50:39 DEBUG (HttpSymbolServer) HttpSymbolServer[https://msdl.microsoft.com/download/symbols/]: Getting file [ExplorerExtensions.pdb/CD9E6CA1215B466A9CA7D946299DE6E11/ExplorerExtensions.pdb]: https://msdl.microsoft.com/download/symbols/ExplorerExtensions.pdb/CD9E6CA1215B466A9CA7D946299DE6E11/ExplorerExtensions.pdb GET  
2021-07-03 23:50:39 DEBUG (HttpSymbolServer) HttpSymbolServer[https://msdl.microsoft.com/download/symbols/]: Http response: 200  
2021-07-03 23:50:45 ERROR (LoadPdbDialog) Error Getting Symbol File: java.io.IOException: closed  
```

**Comment by dev747368** [comment]
Do you have full access to "C:\ProgramData\Dbg\sym"?  What else is in that directory?

What was the full path where you saw the pdb.tmp file?

Could you try choosing a new & empty local symbol storage directory under your c:\users\yourusername directory and see if the download fails there?

**Comment by justanotheranonymoususer** [comment]
> Do you have full access to "C:\ProgramData\Dbg\sym"?

Yes. It is being created by WinDbg Preview with full access to all apps.

> What else is in that directory?

Symbols and an empty `pingme.txt`.

> What was the full path where you saw the pdb.tmp file?

`C:\ProgramData\Dbg\sym\ExplorerExtensions.pdb\CD9E6CA1215B466A9CA7D946299DE6E11`

> Could you try choosing a new & empty local symbol storage directory under your c:\users\yourusername directory and see if the download fails there?

I tried with `C:\Users\myusername\Desktop\sym`. Same error.

Note that I used `C:\ProgramData\Dbg\sym` with Ghidra 9 and other tools and never had issues.

**Comment by dev747368** [comment]
Were there 000admin directories in the original and the new directory?

**Comment by justanotheranonymoususer** [comment]
In `C:\ProgramData\Dbg\sym` - no. In `C:\Users\myusername\Desktop\sym` - yes, I said yes when Ghidra asked about creating a Microsoft symbol folder structure.

**Comment by dev747368** [comment]
Good news!  I figured out the problem.... your internet is just too slow...  😄 

Well, actually, there is a timeout that is too short on the connection.  Any download that takes more than 10 seconds is getting interrupted by a timer.

I'll update this ticket when a fix is going in.

**Comment by justanotheranonymoususer** [comment]
I experience the exact same issue in 10.0.2 :(

**Comment by ryanmkurtz** [comment]
> I experience the exact same issue in 10.0.2 :(
> 
> Log: https://i.imgur.com/JFBxNqR.png

This has been fixed for the upcoming 10.1 release.  10.0.2 is expected to have the problem.
